rpm-rs doesn't build with some features turned off

It should be possible to compile the codebase with `cargo build --no-default-features`. After #1 is merged, we still have the following errors:

```
error[E0599]: no function or associated item named `builder` found for struct `header::Header` in the current scope
   --> src/rpm/builder.rs:304:58
    |
304 |         let digest_header = Header::<IndexSignatureTag>::builder()
    |                                                          ^^^^^^^ function or associated item not found in `header::Header<constants::IndexSignatureTag>`
    |
   ::: src/rpm/headers/header.rs:38:1
    |
38  | pub struct Header<T: num::FromPrimitive> {
    | ---------------------------------------- function or associated item `builder` not found for this

warning: unused import: `std::io::Read`
  --> src/rpm/package.rs:17:5
   |
17 | use std::io::Read;
   |     ^^^^^^^^^^^^^
   |
   = note: `#[warn(unused_imports)]` on by default
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -56,6 +56,25 @@ jobs:
         with:
           command: test
 
+  test-no-default-features:
+    name: Test Suite (no-default-features)
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        rust:
+          - stable
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.rust }}
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: --no-default-features
+
   fmt:
     name: Rustfmt
     runs-on: ubuntu-latest

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -7,6 +7,7 @@
 //!
 //! ```rust
 //!
+//! # #[cfg(feature = "signature-meta")]
 //! use rpm::{
 //!     signature::pgp::{
 //!         Signer,
@@ -16,6 +17,8 @@
 //! use std::str::FromStr;
 //!
 //! # fn main() -> Result<(), Box<dyn std::error::Error>> {
+//! # #[cfg(feature = "signature-meta")]
+//! # {
 //! let raw_secret_key = std::fs::read("./test_assets/secret_key.asc")?;
 //! let pkg = rpm::RPMBuilder::new("test", "1.0.0", "MIT", "x86_64", "some awesome package")
 //!             .compression(rpm::Compressor::from_str("gzip")?)
@@ -53,6 +56,7 @@
 //! let pkg = rpm::RPMPackage::parse(&mut buf_reader)?;
 //! // verifying
 //! pkg.verify_signature(Verifier::load_from_asc_bytes(&raw_pub_key)?)?;
+//! # }
 //! # Ok(())
 //! # }
 //! ```

--- a/src/rpm/builder.rs
+++ b/src/rpm/builder.rs
@@ -301,12 +301,17 @@ impl RPMBuilder {
 
         let header_and_content_len = header.len() + content.len();
 
-        let digest_header = Header::<IndexSignatureTag>::builder()
-            .add_digest(
-                header_digest_sha1.as_str(),
-                header_and_content_digest_md5.as_slice(),
-            )
-            .build(header_and_content_len as i32);
+        #[cfg(feature = "signature-meta")]
+        let digest_header = {
+            Header::<IndexSignatureTag>::builder()
+                .add_digest(
+                    header_digest_sha1.as_str(),
+                    header_and_content_digest_md5.as_slice(),
+                )
+                .build(header_and_content_len as i32)
+        };
+        #[cfg(not(feature = "signature-meta"))]
+        let digest_header = { Header::<IndexSignatureTag>::new_empty() };
 
         let metadata = RPMPackageMetadata {
             lead,

--- a/src/rpm/headers/header.rs
+++ b/src/rpm/headers/header.rs
@@ -126,6 +126,7 @@ where
             store,
         })
     }
+
     pub(crate) fn parse<I: std::io::BufRead>(input: &mut I) -> Result<Header<T>, RPMError> {
         let mut buf: [u8; 16] = [0; 16];
         input.read_exact(&mut buf)?;
@@ -389,6 +390,14 @@ impl Header<IndexSignatureTag> {
     #[inline]
     pub fn get_file_ima_signature_length(&self) -> Result<i32, RPMError> {
         self.get_entry_i32_data(IndexSignatureTag::RPMSIGTAG_FILESIGNATURE_LENGTH)
+    }
+
+    pub fn new_empty() -> Self {
+        Self {
+            index_header: IndexHeader::new(0, 0),
+            index_entries: vec![],
+            store: vec![],
+        }
     }
 
     pub fn clear(&mut self) {
@@ -677,6 +686,7 @@ where
 mod tests2 {
     use super::*;
 
+    #[cfg(feature = "signature-meta")]
     #[test]
     fn signature_header_build() {
         let size: i32 = 209_348;

--- a/src/tests.rs
+++ b/src/tests.rs
@@ -1,4 +1,5 @@
 use super::*;
+#[cfg(feature = "signature-meta")]
 use crate::signature::pgp::{Signer, Verifier};
 
 fn test_private_key_path() -> std::path::PathBuf {
@@ -29,6 +30,7 @@ fn cargo_manifest_dir() -> std::path::PathBuf {
     std::path::PathBuf::from(env!("CARGO_MANIFEST_DIR"))
 }
 
+#[cfg(feature = "signature-meta")]
 #[test]
 fn test_rpm_file_signatures() -> Result<(), Box<dyn std::error::Error>> {
     let rpm_file_path = file_signatures_test_rpm_file_path();
@@ -54,6 +56,7 @@ fn test_rpm_file_signatures() -> Result<(), Box<dyn std::error::Error>> {
     Ok(())
 }
 
+#[cfg(feature = "signature-meta")]
 #[test]
 fn test_rpm_file_signatures_resign() -> Result<(), Box<dyn std::error::Error>> {
     let rpm_file_path = file_signatures_test_rpm_file_path();
@@ -303,6 +306,7 @@ fn test_rpm_header_base(package: RPMPackage) -> Result<(), Box<dyn std::error::E
     Ok(())
 }
 
+#[cfg(feature = "async-tokio")]
 #[tokio::test]
 async fn test_rpm_header_async() -> Result<(), Box<dyn std::error::Error>> {
     let rpm_file_path = test_rpm_file_path();

